### PR TITLE
VTCCA-2931 - test coverage - 2nd instalment

### PR DESCRIPTION
**Pull request by PeterPerhac**
85.38%

### CODE REVIEW

File: app/uk/gov/hmrc/voapropertylinking/connectors/modernised/MdtpDashboardManagementApi.scala [hunk 1]
```diff
@@ -45,7 +45,12 @@ class MdtpDashboardManagementApi @Inject()(
 
   def getAssessment(authorisationId: Long)(implicit hc: HeaderCarrier): Future[Option[PropertiesView]] = {
     val url = baseUrl + s"/mdtp-dashboard-management-api/mdtp_dashboard/view_assessment?listYear=$listYear&authorisationId=$authorisationId"
-    http.GET[Option[PropertiesView]](url).map(_.map(_.upperCase)) recover { case _: NotFoundException => None }
+
+    http.GET[Option[PropertiesView]](url).map { optPropertiesView =>
+      PartialFunction.condOpt(optPropertiesView) {
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/voapropertylinking/connectors/modernised/MdtpDashboardManagementApi.scala]: does this compile??? from the look of it, it looks like itll be a future option of an option

File: project/MicroService.scala [hunk 2]
```diff
@@ -39,7 +39,7 @@ trait MicroService {
     // Semicolon-separated list of regexs matching classes to exclude
     import scoverage.ScoverageKeys
     Seq(
-      ScoverageKeys.coverageExcludedPackages := "<empty>;Reverse.*;views.*;uk.gov.hmrc.voapropertylinking.config.*;.*\\.test\\..*;poc.view.*;poc.uk.gov.hmrc.voapropertylinking.config.*;.*(AuthService|BuildInfo|Routes).*",
+      ScoverageKeys.coverageExcludedPackages := "<empty>;Reverse.*;views.*;uk.gov.hmrc.voapropertylinking.config.*;.*\\.temp\\..*;.*\\.test\\..*;poc.view.*;poc.uk.gov.hmrc.voapropertylinking.config.*;.*(AuthService|BuildInfo|Routes).*",
       ScoverageKeys.coverageMinimum := 50,
```
Comment (DanielConnelly) [inline project/MicroService.scala]: change minimum to 80 and turn on fail.

File: test/binders/propertylinks/GetMyOrganisationPropertyLinksParametersSpec.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import binders._
+import cats.data.Validated.Valid
+
+class GetMyOrganisationPropertyLinksParametersSpec extends BaseUnitSpec {
+
+  import GetMyOrganisationPropertyLinksParameters._
+
+  "validating GetMyOrganisationPropertyLinksParameters" should {
+    "come out VALID" when {
+      "no parameters are provided because all parameters are optional" in {
```
Comment (DanielConnelly) [inline test/binders/propertylinks/GetMyOrganisationPropertyLinksParametersSpec.scala]: are all parameters really optional :O

File: test/controllers/PropertyLinkingBaseControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    val function: Foo => Future[Result] =
+      _ => Future.successful(Ok)
+
+    val foo: Foo = Foo(bar = "FooName")
+    val fooJson: JsValue = Json.toJson(foo)
+    val barJson: JsValue = Json.obj()
+  }
+
+  "withJsonBody" should {
+    "evaluate function body" when {
```
Comment (DanielConnelly) [inline test/controllers/PropertyLinkingBaseControllerSpec.scala]: is this not testing logic from a library/

File: test/repositories/DVRRepositorySpec.scala [hunk 5]
```diff
@@ -30,16 +30,13 @@ import scala.concurrent.ExecutionContext.Implicits.global
 class DVRRepositorySpec
   extends BaseUnitSpec
     with MongoSpecSupport {
-  val app = appBuilder.build()
 
-  val repository = new DVRRepository(app.injector.instanceOf[ReactiveMongoComponent], "test", app.injector.instanceOf[ServicesConfig]) {
-    override def indexes: Seq[Index] = Seq.empty
-  }
+  val app = new GuiceApplicationBuilder()
```
Comment (DanielConnelly) [inline test/repositories/DVRRepositorySpec.scala]: can we label this one for a move to IT spec when it is valid.

**Comment by DanielConnelly** [comment]
your stuff looks good, this service defo need some TLC, but a few comments but looks good.

File: test/binders/propertylinks/GetMyOrganisationPropertyLinksParametersSpec.scala (hunk 3 above)
Comment (PeterPerhac) [reply test/binders/propertylinks/GetMyOrganisationPropertyLinksParametersSpec.scala]: indeed, surprised me but it is what it is, this is how i found it

File: test/controllers/PropertyLinkingBaseControllerSpec.scala (hunk 4 above)
Comment (PeterPerhac) [reply test/controllers/PropertyLinkingBaseControllerSpec.scala]: no, we don't use the library stuff, we have our own withJsonBody in the base controller, and wasn't covered by tests

File: app/uk/gov/hmrc/voapropertylinking/connectors/modernised/MdtpDashboardManagementApi.scala (hunk 1 above)
Comment (PeterPerhac) [reply app/uk/gov/hmrc/voapropertylinking/connectors/modernised/MdtpDashboardManagementApi.scala]: i am so puzzled by this. I think this has gone wrong in the merge, i clicked the magic wand and this happened. I am going to sort this out

Comment (PeterPerhac) [reply app/uk/gov/hmrc/voapropertylinking/connectors/modernised/MdtpDashboardManagementApi.scala]: yes, this definitely is a bad merge issue... thanks for paying attention 👍

**Comment by DanielConnelly** [comment]
I am happy with this LGTM 👍

**Pull request merged**